shell/platform/darwin/ios/framework/Source/FlutterTextEditingDelta.mm

```diff
@@ -0,0 +1,43 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterTextEditingDelta.h"
+
+@implementation FlutterTextEditingDelta
+
+- (instancetype)initTextEditingDelta:(NSString*)textBeforeChange
+                       replacedRange:(NSRange)range
+                         updatedText:(NSString*)text {
+  self = [super init];
+
+  if (self) {
+    NSInteger start = range.location;
+    NSInteger end = range.location + range.length;
+    [self setDeltas:textBeforeChange newText:text deltaStart:start deltaEnd:end];
+  }
+
+  return self;
+}
+
+- (instancetype)initWithNonText:(NSString*)text {
+  self = [super init];
+
+  if (self) {
+    [self setDeltas:text newText:@"" deltaStart:-1 deltaEnd:-1];
+  }
+
+  return self;
+}
+
+- (void)setDeltas:(NSString*)oldText
+          newText:(NSString*)newTxt
+       deltaStart:(NSInteger)newStart
+         deltaEnd:(NSInteger)newEnd {
+  _oldText = oldText;
```

Comment:
Since this method is only called from the constructors of this class is this alright? 

https://github.com/flutter/engine/blob/45dc2fee1304805f54d72563f49873e226e4cd95/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm#L594 Here the general pattern seems to set the ivars in the constructor.

Reply:
Those are new objects created in the initializer and here you're potentially transferring the object's ownership. You'll have to retain them manually if you want to set them directly to ivars.

Reply:
Oh since the delta objects are immediately turned into json, not claiming the ownership of these strings should be fine.